R/ROOT/build_tarballs.jl

```diff
@@ -0,0 +1,216 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+
+using BinaryBuilder, Pkg
+
+name = "ROOT"
+version = v"6.32.8"
+
+rootgithash = Dict(v"6.32.6" => "5380676d1f0ed3055048f305de917572f976a090",
+                   v"6.32.8" => "a1748a069d2d5f27ec528cfd89b54b12dcba793e")
+
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/root-project/root.git", rootgithash[version]),
+    DirectorySource("./bundled")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE
+if echo "$target" | grep -q musl; then #build wih musl library
+# Disabling what is not working with musl:
+    CMAKE_EXTRA_OPTS=(-Ddavix=OFF -Dxrootd=OFF -Dssl=OFF -Druntime_cxxmodules=OFF -Droofit=OFF)
+else
+    CMAKE_EXTRA_OPTS=(-Druntime_cxxmodules=OFF)
+fi
+
+#Uncomment for a minimal build for debugging purposes
+#CMAKE_EXTRA_OPTS+=(-Dclad=OFF -Dhtml=OFF -Dwebgui=OFF -Dcxxmodules=OFF -Dproof=OFF -Dtmva=OFF -Drootfit=OFF -Dxproofd=OFF -Dxrootd=OFF -Dssl=OFF -Dpyroot=OFF -Dtesting=OFF -Droot7=OFF -Dspectrum=OFF -Dunfold=OFF -Dasimage=OFF -Dgviz=OFF -Dfitiso=OFF -Dcocoa=OFF -Dopengl=OFF -Dproof=OFF -Dxml=OFF -Dgfal=OFF -Dmpi=OFF)
+
+export SYSTEM_INCLUDE_PATH="`g++ --sysroot="/opt/$target/$target/sys-root" -E -x c++ -v /dev/null  2>&1  | awk '{gsub(\"^ \", \"\")} /End of search list/{a=0} {if(a==1){s=s d $0;d=":"}} /#include <...> search starts here/{a=1} END{print s}'`"
+
+# build-in compilation of the libAfterImage library needs this directory
+mkdir -p /tmp/user/0
+
+cd /
+
+cd "$WORKSPACE"
+
+# For the cross-compiling, LLVM needs to compile the llvm-tblgen tool
+# for the host. The ROOT LLVM code tree does not include the test and
+# benchmark directories cmake will need to configure for the host for
+# this bootstrap. Therefore, we need to disable the build of the test and
+# benchmarks.
+sed -i 's/\(option(LLVM_INCLUDE_TESTS[[:space:]].*[[:space:]]\)on)\(.*\)/\1OFF)\2/i
+        s/\(option(LLVM_INCLUDE_BENCHMARKS[[:space:]].*[[:space:]]\)on)\(.*\)/\1OFF)\2/i' \
+        srcdir/root/interpreter/llvm-project/llvm/CMakeLists.txt
+echo "set(CXX_STANDARD 17)" >> srcdir/root/interpreter/llvm-project/llvm/CMakeLists.txt
+
+# N llvm links. LLVM link command needs 15GB
+njobs=$((2*`nproc`))
```

Comment:
Please use the [environment variable`${nproc}`](https://docs.binarybuilder.org/stable/build_tips/#Automatic-environment-variables), not `$(nproc)`, that's going to kill the machine.
```suggestion
njobs=${nproc}
```

Reply:
I had an issue with `$nproc`. On the cluster I'm using it is set the total number of cores of the node, instead of number of allocate cores, as `nproc` command does. How is it set?
```
# allocate 32 cores on a node
salloc -c 32 -p htc_interactive
srun --pty bash -i
sandbox:${WORKSPACE} # echo $nproc
128
sandbox:${WORKSPACE} # nproc
32
```

I apply the patch and I will use `BINARYBUILDER_NPROC` to override the value.

Factor 2 was to account the build IO limits the CPU load, but I'm fine to remove it.

Reply:
The difference from `$nproc` content and `nproc` command comes from the fact that `Sys.CPU_THREADS` used by `$nproc` does not account for the CPU affinity mask (see `tasket` and `sched_setaffinity`), while the `nproc` command does (see https://github.com/coreutils/gnulib/blob/ac1e6fee24c16c87d488b3d44cf3f474c10dd9bf/lib/nproc.c#L268). 

```
taskset 1 julia -e 'println(Sys.CPU_THREADS, " vs ", readchomp(`nproc`))'
```

displays `8 vs 1` on my 8-thread laptop.